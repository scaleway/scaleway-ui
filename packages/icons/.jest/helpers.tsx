import { ThemeProvider } from '@emotion/react'
import makeHelpers from '@scaleway/jest-helpers'
import type { ReactNode } from 'react'
import { theme } from '@ultraviolet/ui'

type WrapperProps = {
  theme?: typeof theme
  children: ReactNode
}

const Wrapper = ({ theme, children }: WrapperProps) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

export const {
  renderWithTheme,
  shouldMatchEmotionSnapshot,
  shouldMatchEmotionSnapshotWithPortal,
} = makeHelpers(Wrapper, { classNameReplacer: className => className })
