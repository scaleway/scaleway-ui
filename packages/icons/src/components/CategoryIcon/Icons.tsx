import ai from './assets/ai.svg'
import baremetal from './assets/baremetal.svg'
import billing from './assets/billing.svg'
import compute from './assets/compute.svg'
import console from './assets/console.svg'
import containers from './assets/containers.svg'
import database from './assets/database.svg'
import datacenter from './assets/datacenter.svg'
import dedicatedServer from './assets/dedicated-server.svg'
import devTools from './assets/dev-tools.svg'
import documentation from './assets/documentation.svg'
import iot from './assets/iot.svg'
import labs from './assets/labs.svg'
import managedServices from './assets/managed-services.svg'
import network from './assets/network.svg'
import observability from './assets/observability.svg'
import security from './assets/security.svg'
import serverless from './assets/serverless.svg'
import storage from './assets/storage.svg'
import toolsServices from './assets/tools-services.svg'
import useCase from './assets/use-case.svg'
import vpc from './assets/vpc.svg'
import webHosting from './assets/web-hosting.svg'

export const CATEGORY_ICONS = {
  baremetal,
  webHosting,
  vpc,
  useCase,
  toolsServices,
  storage,
  serverless,
  security,
  observability,
  network,
  managedServices,
  iot,
  documentation,
  dedicatedServer,
  datacenter,
  database,
  containers,
  console,
  compute,
  billing,
  ai,
  labs,
  devTools,
} as const
