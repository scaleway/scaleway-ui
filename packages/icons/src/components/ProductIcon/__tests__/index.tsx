import { describe, it } from '@jest/globals'
import { ProductIcon } from '..'
import { shouldMatchEmotionSnapshotWithPortal } from '../../../../.jest/helpers'
import { PRODUCT_ICONS } from '../Icons'

describe('ProductIcon', () => {
  describe('should work with all names', () => {
    Object.keys(PRODUCT_ICONS).forEach(iconName => {
      it(`should work with ${iconName}`, () =>
        shouldMatchEmotionSnapshotWithPortal(<ProductIcon name="console" />))
    })
  })

  it('should work with disabled', () =>
    shouldMatchEmotionSnapshotWithPortal(
      <ProductIcon name="console" disabled />,
    ))

  describe('should work with all sizes', () => {
    ;(['small', 'medium', 'large', 'xlarge'] as const).forEach(size => {
      it(`should work with ${size}`, () =>
        shouldMatchEmotionSnapshotWithPortal(
          <ProductIcon name="console" size={size} />,
        ))
    })
  })

  describe('should work with all variants', () => {
    ;(['primary', 'danger', 'warning'] as const).forEach(variant => {
      it(`should work with ${variant}`, () =>
        shouldMatchEmotionSnapshotWithPortal(
          <ProductIcon name="console" variant={variant} />,
        ))
    })
  })
})
