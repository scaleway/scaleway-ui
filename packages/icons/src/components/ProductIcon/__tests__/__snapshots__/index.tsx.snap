// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductIcon should work with all names should work with abuse 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with advancedSettings 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with apiGateway 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with apiKey 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with appleSilicon 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with application 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with applicationLibrary 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with backends 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with basicSupport 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with billing 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with blockStorage 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with cb 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with cdn 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with changelog 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with cli 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with cockpit 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with console 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with containers 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with dedibox 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with dedicatedServer 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with devices 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with directConnect 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with dns 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with documentDB 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with documentation 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with domains 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with elasticMetal 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with file 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with flexibleIp 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with folder 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with frontends 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with functions 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with goldSupport 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with hubNetworks 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with hubRoutes 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with iam 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with images 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with instance 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with instanceEnterprise 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with instanceGpu 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with iot 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with iotEdge 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with ipFailover 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with ipfs 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with ipfsNaming 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with k8sKosmos 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with kubernetes 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with lb 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with lifeCycleRules 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with macMiniM2 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with multiUser 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with nats 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with objectStorage 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with packer 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with placementGroup 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with platinumSupport 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with policy 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with pool 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with postgreSqlAndMySql 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with privateNetwork 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with publicGateway 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with queueing 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with rabbitMQ 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with rdb 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with redis 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with registry 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with rocket 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with secretManager 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with securityGroup 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with sepa 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with server 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with serverlessDB 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with serverlessJobs 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with silverSupport 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with sms 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with smtp 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with snapshots 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with sns 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with sqs 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with sslCertificates 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with support 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with terraform 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with transactionalEmail 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with tutorial 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with user 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with verifyCard 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with video 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with volume 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with vpc 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with webhosting 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all names should work with zone 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all sizes should work with large 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-1cn9fyi-StyledIcon euby9wl0"
    size="large"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all sizes should work with medium 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-7ewfw6-StyledIcon euby9wl0"
    size="medium"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all sizes should work with small 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all sizes should work with xlarge 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-hy5q6k-StyledIcon euby9wl0"
    size="xlarge"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all variants should work with danger 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-1a6tz7p-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all variants should work with primary 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-yif6bn-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with all variants should work with warning 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-1fvi5o-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;

exports[`ProductIcon should work with disabled 1`] = `
<DocumentFragment>
  <test-file-stub
    classname="cache-ovorz1-StyledIcon euby9wl0"
    size="small"
    viewbox="0 0 64 64"
  >
    test-file-stub
  </test-file-stub>
</DocumentFragment>
`;
